ui/app/adapters/watchable.js

```diff
@@ -62,6 +85,47 @@ export default ApplicationAdapter.extend({
     });
   },
 
+  query(store, type, query, snapshotRecordArray, options, additionalParams = {}) {
+    const params = assign({}, query, additionalParams);
+    const url = this.buildURL(type.modelName, null, null, 'query', query);
+
+    if (get(options, 'adapterOptions.watch')) {
+      // The intended query without additional blocking query params is used
+      // to track the appropriate query index.
+      params.index = this.watchList.getIndexFor(`${url}?${queryString.stringify(query)}`);
+    }
+
+    const abortToken = get(options, 'adapterOptions.abortToken');
+    return this.ajax(url, 'GET', {
+      abortToken,
+      data: params,
+    }).then(payload => {
+      const adapter = store.adapterFor(type.modelName);
+
+      // Query params may not necessarily map one-to-one to attribute names.
+      // Adapters are responsible for declaring param mappings.
+      const queryParamsToAttrs = Object.keys(adapter.queryParamsToAttrs).map(key => ({
```

Comment:
Yikes, good catch! It can't happen today because this code path will only ever happen for volumes, but it should protect against a null value either way.